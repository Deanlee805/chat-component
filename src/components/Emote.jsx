import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

const EmoteBox = styled.span`
  height: 38px;
  width: 38px;
  display: block;
  text-align: center;
  &:hover {
    background-color: rgba(125,91,190,.2);
    cursor: pointer;
  }
`;

const EmoteImg = styled.img`
  padding-top: 4px;
  max-height: 28px;
  max-width: 28px;
`;

export const Emote = (props) => {
  return (
    <EmoteBox onClick={() => props.emoteClicked(props.phrase)}>
      <EmoteImg src={props.src} />
    </EmoteBox>
  );
};