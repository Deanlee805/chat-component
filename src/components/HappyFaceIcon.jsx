import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

const IconContainer = styled.span`
  position: absolute;
  right: 4px;
  top: 4px;
  height: 21px;
  width: 21px;
  border: 1px solid transparent;
  transition: all 0.1s ease;
  &:hover {
    background: rgba(100,65,164,0.5);
    border: 1px solid transparent;
    border-radius: 4px;
    cursor: pointer;
    opacity: 0.3;
  }
`;

const Path = styled.path`
  fill: #6441a4;
`;

export const HappyFaceIcon = (props) => {
  return (
    <IconContainer onClick={props.clickHandler}>
      <svg
        width="20px"
        height="20px"
        version="1.1"
        viewBox="0 0 20 20"
        x="0px"
        y="0px"
      >
        <Path
          d="M10 1a9 9 0 0 0-9 9 9 9 0 0 0 9 9 9 9 0 0 0 9-9 9 9 0 0 0-9-9m0 2c3.859 0 7 3.141 7 7s-3.141 7-7 7-7-3.141-7-7 3.141-7 7-7zm4 6a1 1 0 1 0-2 0 1 1 0 1 0 2 0zM8 9a1 1 0 1 0-2 0 1 1 0 1 0 2 0zm4.571 3H7.428c-.237 0-.428.168-.428.375C7 13.822 8.345 15 10 15c1.654 0 3-1.178 3-2.625 0-.207-.192-.375-.429-.375z"
          fill-rule="evenodd"
        />
      </svg>
    </IconContainer>
  );
};