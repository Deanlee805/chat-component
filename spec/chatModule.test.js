import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ChatContainer } from '../src/components/ChatContainer';
import ChatBox from '../src/components/ChatBox';
import { Chat } from '../src/components/Chat';
import { PostMessageBox } from '../src/components/PostMessageBox';


describe('ChatContainer tests', () => {
  it ('ChatContainer render correctly', () => {
    const tree = renderer
      .create(<ChatContainer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ChatContainer render ChatBox ', () => {
    const wrapper = shallow(<ChatContainer />);
    expect(wrapper.find('ChatBox')).not.toBeNull();
  });
});


describe('ChatBox tests', () => {
  const twitchChats = [
    {
      id: 9,
      username: 'chiquita_bananna',
      twitch_sub: true,
      mod_status: true
    },
    {
      id: 502,
      username: 'A_Seagull',
      twitch_sub: true,
      mod_status: true
    },
  ];

  test('ChatBox render correctly', () => {
    const component = renderer.create(<ChatBox chatsArray={twitchChats}/>);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('ChatBox should render a Chat component for each chatObject passed to it', () => {
    const wrapper = mount(<ChatBox chatsArray={twitchChats}/>);
    expect(wrapper.find(Chat)).toHaveLength(2);
  });
});

describe('Chat tests', () => {
  const user1 = {
    id: 9,
    username: 'chiquita_bananna',
    twitch_sub: true,
    mod_status: true
  };

  const user2 = {
    id: 503,
    username: 't3nwinter',
    twitch_sub: true,
    mod_status: false,
  };

  const user3 = {
    id: 484,
    username: 'realmgamingg',
    twitch_sub: false,
    mod_status: false,
  };

  test('Chat render correctly', () => {
    const component = renderer.create(<Chat chat={user1}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Chat should render appropriate mod or sub icons per user\'s mod and sub status', () => {
    const modAndSubWrapper = shallow(<Chat chat={user1} />);
    const onlySubWrapper = shallow(<Chat chat={user2} />);
    const normalUserWrapper = shallow(<Chat chat={user3} />);

    expect(modAndSubWrapper.find('.mod').length && modAndSubWrapper.find('.sub').length).toBe(1);
    expect(onlySubWrapper.find('.sub').length).toBe(1);
    expect(onlySubWrapper.find('.mod').length).toBe(0);
    expect(normalUserWrapper.find('.mod').length && normalUserWrapper.find('.sub').length).toBe(0);
  });
});

describe('PostMessageBox tests', () => {
  it('PostMessageBox render correctly', () => {
    const tree = renderer
      .create(<PostMessageBox />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});




