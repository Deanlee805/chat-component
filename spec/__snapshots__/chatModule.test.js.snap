// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat tests Chat render correctly 1`] = `
<div>
  <span />
   
   
  <span>
    <img
      className="mod"
      height="18px"
      src="https://s3-us-west-1.amazonaws.com/twitchchat/mod.png"
      width="18px"
    />
  </span>
  <span>
    <img
      className="sub"
      height="18px"
      src="https://s3-us-west-1.amazonaws.com/twitchchat/sub.png"
      width="18px"
    />
  </span>
  <span
    className="Chat__Username-sc-11hxeqq-0 jwgTiW"
  >
     
    chiquita_bananna
  </span>
  :
   
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
</div>
`;

exports[`ChatBox tests ChatBox render correctly 1`] = `
<div
  className="ChatBox__ChatBoxStyle-sc-8d4jv0-0 exzOIu"
  onScroll={[Function]}
>
  <div>
    <span />
     
     
    <span>
      <img
        className="mod"
        height="18px"
        src="https://s3-us-west-1.amazonaws.com/twitchchat/mod.png"
        width="18px"
      />
    </span>
    <span>
      <img
        className="sub"
        height="18px"
        src="https://s3-us-west-1.amazonaws.com/twitchchat/sub.png"
        width="18px"
      />
    </span>
    <span
      className="Chat__Username-sc-11hxeqq-0 jwgTiW"
    >
       
      chiquita_bananna
    </span>
    :
     
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
  <div>
    <span />
     
     
    <span>
      <img
        className="mod"
        height="18px"
        src="https://s3-us-west-1.amazonaws.com/twitchchat/mod.png"
        width="18px"
      />
    </span>
    <span>
      <img
        className="sub"
        height="18px"
        src="https://s3-us-west-1.amazonaws.com/twitchchat/sub.png"
        width="18px"
      />
    </span>
    <span
      className="Chat__Username-sc-11hxeqq-0 jwgTiW"
    >
       
      A_Seagull
    </span>
    :
     
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`ChatContainer tests ChatContainer render correctly 1`] = `
<div
  className="ChatContainer__App-hd0om2-0 ccIdBA"
>
  <div
    className="ChatContainer__Header-hd0om2-1 dXMVrm"
  >
    Chat On Videos
  </div>
  <div
    className="ChatBox__ChatBoxStyle-sc-8d4jv0-0 exzOIu"
    onScroll={[Function]}
  />
  <div
    className="PostMessageBox__TextBoxContainer-f9nk4c-0 hqfwsX"
  >
    <div
      className="PostMessageBox__TextBox-f9nk4c-1 cwOWzG"
    >
      <textarea
        className="PostMessageBox__TextArea-f9nk4c-2 TEyAC"
        id="textBox"
        placeholder="Post a message"
      />
      <div>
        <img />
      </div>
    </div>
    <div
      className="PostMessageBox__PostButtonContainer-f9nk4c-3 gPnsfj"
    >
      <button
        className="PostMessageBox__PostButton-f9nk4c-4 FsBuX"
        id="postButton"
        onClick={[Function]}
      >
        Chat
      </button>
    </div>
  </div>
</div>
`;

exports[`PostMessageBox tests PostMessageBox render correctly 1`] = `
<div
  className="PostMessageBox__TextBoxContainer-f9nk4c-0 hqfwsX"
>
  <div
    className="PostMessageBox__TextBox-f9nk4c-1 cwOWzG"
  >
    <textarea
      className="PostMessageBox__TextArea-f9nk4c-2 TEyAC"
      id="textBox"
      placeholder="Post a message"
    />
    <div>
      <img />
    </div>
  </div>
  <div
    className="PostMessageBox__PostButtonContainer-f9nk4c-3 gPnsfj"
  >
    <button
      className="PostMessageBox__PostButton-f9nk4c-4 FsBuX"
      id="postButton"
      onClick={[Function]}
    >
      Chat
    </button>
  </div>
</div>
`;
